openassessment/assessment/test/test_staff.py

```diff
@@ -0,0 +1,348 @@
+                    "explanation": "Happy to give it up."
+                }
+            ]
+        },
+        {
+            "name": "singing",
+            "prompt": "Did the writer break into tedious elvish lyrics?",
+            "options": [
+                {"name": "no", "points": "2", "explanation": ""},
+                {"name": "yes", "points": "0", "explanation": ""}
+            ]
+        },
+    ]
+}
+
+# Answers are against RUBRIC_DICT -- this is worth 6 points
+ASSESSMENT_DICT = {
+    'overall_feedback': u"这是中国",
+    'criterion_feedback': {
+        "giveup": u"𝓨𝓸𝓾 𝓼𝓱𝓸𝓾𝓵𝓭𝓷'𝓽 𝓰𝓲𝓿𝓮 𝓾𝓹!"
+    },
+    'options_selected': {
+        "secret": "yes",
+        u"ⓢⓐⓕⓔ": "no",
+        "giveup": "reluctant",
+        "singing": "no",
+    },
+}
+
+# Answers are against RUBRIC_DICT -- this is worth 0 points
+ASSESSMENT_DICT_FAIL = {
+    'overall_feedback': u"fail",
+    'criterion_feedback': {},
+    'options_selected': {
+        "secret": "no",
+        u"ⓢⓐⓕⓔ": "no",
+        "giveup": "unwilling",
+        "singing": "yes",
+    }
+}
+
+# Answers are against RUBRIC_DICT -- this is worth 14 points
+ASSESSMENT_DICT_PASS = {
+    'overall_feedback': u"这是中国",
+    'criterion_feedback': {},
+    'options_selected': {
+        "secret": "yes",
+        u"ⓢⓐⓕⓔ": "yes",
+        "giveup": "eager",
+        "singing": "no",
+    }
+}
+
+# Answers are against RUBRIC_DICT -- this is worth 14 points
+# Feedback text is one character over the limit.
+LONG_FEEDBACK_TEXT = u"是" * Assessment.MAX_FEEDBACK_SIZE + "."
+ASSESSMENT_DICT_HUGE = {
+    'overall_feedback': LONG_FEEDBACK_TEXT,
+    'criterion_feedback': {
+        "secret": LONG_FEEDBACK_TEXT,
+        u"ⓢⓐⓕⓔ": LONG_FEEDBACK_TEXT,
+        "giveup": LONG_FEEDBACK_TEXT,
+        "singing": LONG_FEEDBACK_TEXT,
+    },
+    'options_selected': {
+        "secret": "yes",
+        u"ⓢⓐⓕⓔ": "yes",
+        "giveup": "eager",
+        "singing": "no",
+    },
+}
+
+
+
+@ddt
+class TestStaffOverwrite(CacheResetTest):
+    """
+    Tests for staff assessments made as overrides, when none is required to exist.
+    """
+
+    STEPS = ['self']
+    STEP_REQUIREMENTS = {}
+    STEP_REQUIREMENTS_WITH_STAFF = {'staff': {'required': True}}
+
+    def test_create_assessment_points(self):
+        # Create assessment
+        tim_sub, tim = self.create_student_and_submission("Tim", "Tim's answer")
+
+        # Staff assess it
+        assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            ASSESSMENT_DICT['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+
+        # Ensure points are calculated properly
+        self.assertEqual(assessment["points_earned"], 6)
+        self.assertEqual(assessment["points_possible"], 14)
+
+        #ensure submission is marked as finished
+        self.assertTrue(staff_api.assessment_is_finished(tim_sub["uuid"], self.STEP_REQUIREMENTS))
+
+    def test_create_assessment_overrides(self):
```

Comment:
I'd like to see a test with a double staff override, to verify that the latest override beats out all others. The subsequent override should provide different scores to make sure that it is a true override, and that the staff scores don't get aggregated together in any way.